uefi/region_test.go

```diff
@@ -0,0 +1,61 @@
+package uefi
+
+import "testing"
+
+var regionTestcases = [...]struct {
+	in    *Region
+	valid bool
+	base  uint32
+	end   uint32
+}{
+	// Invalid
+	{
+		in:    &Region{0, 0},
+		valid: false,
+		base:  0,
+		end:   0x1000,
+	},
+	{
+		in:    &Region{1, 0},
+		valid: false,
+		base:  0x1000,
+		end:   0x1000,
+	},
+	// Valid
+	{
+		in:    &Region{1, 1},
+		valid: true,
+		base:  0x1000,
+		end:   0x2000,
+	},
+	{
+		in:    &Region{100, 200},
+		valid: true,
+		base:  0x64000,
+		end:   0xC9000,
+	},
+}
```

Comment:
Good stuff, one suggestion: add a test where the limit is right at the largest possible value.

Reply:
Done.